app/src/main/java/com/sun/basic_japanese/kanji_basic/KanjiBasicRecyclerAdapter.kt

```diff
@@ -0,0 +1,72 @@
+package com.sun.basic_japanese.kanji_basic
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.sun.basic_japanese.R
+import com.sun.basic_japanese.base.BaseViewHolder
+import com.sun.basic_japanese.data.model.KanjiBasic
+import com.sun.basic_japanese.util.Constants
+import com.sun.basic_japanese.util.Extensions.setHtmlText
+import com.sun.basic_japanese.util.Extensions.setImage
+import kotlinx.android.synthetic.main.item_kanji_basic.view.*
+
+class KanjiBasicRecyclerAdapter(
+    private val listener: OnKanjiBasicItemClickListener,
+    private var kanjiBasicList: List<KanjiBasic> = mutableListOf()
+) : RecyclerView.Adapter<KanjiBasicRecyclerAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, type: Int): ViewHolder {
+        val itemView =
+            LayoutInflater.from(parent.context).inflate(R.layout.item_kanji_basic, parent, false)
+        return ViewHolder(itemView, listener)
+    }
+
+    override fun getItemCount(): Int = kanjiBasicList.size
+
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        viewHolder.onBindData(kanjiBasicList[position])
+    }
+
+    fun setKanjiBasicList(kanjiBasicList: List<KanjiBasic>) {
+        this.kanjiBasicList = kanjiBasicList
```

Comment:
Dùng MutableList để có thể update lại data theo cách này.
```suggestion
        this.kanjiBasicList.run {
            clear()
            addAll(kanjiBasicList)
        }
```

Reply:
Tìm hiểu về DiffUtil.